docs/user/native.rst

```diff
@@ -84,4 +84,137 @@ transparent to the library user.
+library can avoid the error described above. Use the following procedure to
+implement this feature.
+
+1. Add the following content to your new ``scala-native.properties`` file
+desdribed above. For the purposes of this example assume the library is ``z``.
+Note that if your library has more that one library you can add a comma
+delimited list of libraries. If desired, the comments are not needed.
+
+.. code-block:: properties
+
+    # configuration for glue code
+    # defines SCALANATIVE_LINK_Z if @link("z") annnotation is used (found in NIR)
+    # libraries used, comma delimited
+    nir.link.names = z
+
+2. Now in your native "glue" code add the following. The macro is named
+``SCALANATIVE_LINK_`` plus the uppercased name of the library.
+
+.. code-block:: c
+
+    #ifdef SCALANATIVE_LINK_Z
+
+    #include <zlib.h>
+
+    int scalanative_z_no_flush() { return Z_NO_FLUSH; }
+    // other functions
+
+    #endif
+
+The feature works by querying the NIR code to see if the user code is using the
+``z`` library. If used, ``-DSCALANATIVE_LINK_Z`` is passed to the compiler
+and your "glue" code is then compiled. Otherwise, the macro keeps the code
+inside from compiling. The project dependencies with native code are compiled
+individually so this feature only applies to the current library being compiled.
+
+Adding defines to your library when code is being compiled
+----------------------------------------------------------
+
+If your library requires a C preprocessor define then use this feature to add
+the define ``-DMY_DEFINE`` for example to the options passed to the compiler.
+
+.. code-block:: properties
+
+    # add defines, do not add -D
+    preprocessor.defines = MY_DEFINE, MY_VALUE=2
+
+Add extra include paths for your library
+----------------------------------------
+
+Currently, the native code compilation gathers your files and provides an
+include to you project ``resources/scala-native`` directory. This means
+that code needs to use relative includes. e.g. ``#include "mylib.h"``
+
+This feature allows you to vendor code, include code as is, that has
+system includes. e.g. ``#include <libunwind.h>`` add each segment
+starting from the ``scala-native`` path shown above.
+Additional paths are added starting at ``1`` and must be contiguous.
+
+.. code-block:: properties
+
+    # path to vendored libunwind
+    compile.include.path = platform, posix, libunwind
+    # path to gc base dir
+    compile.include.path1 = gc
+
+Additional paths are added starting at ``1`` and must be contiguous.
+
+Add unique identity to your library for debugging
+-------------------------------------------------
+
+Since these features can apply to libraries that are published to Maven
+Central those coordinates can be used to identify your library. The
+example here is for a Scala Native ``javalib`` library.
+
+.. code-block:: properties
+
+    # output via debugging
+    project.groupId = org.scala-native
+    project.artifactId = javalib
```

Comment:
If I understand correctly this is only used to optionally find the native sources which have hard coded behaviour. In our cases it's nativelib which has special handling in the build, becouse we need to pass the information about the selected gc. This method should be more reliable then scanning all native artifacts.
I think that for general purpouse the coordinates would not be required. Because these are used only internally, maybe we can ommit documenting this behaviour?

Reply:
This is not used except to see the output via debugging. The only thing this has to do with Maven is that most libraries will be published there and that is how the dependency resolver works with these coordinates so it is familiar and a decent way to id the library being compiled. I am not sure if it is useful for other than debugging but this may be a nice way to help identify problems that someone may have with their build.

Reply:
> This is not used except to see the output via debugging.

Why not just log the relevant Jar file? This is no different than how the Scala compiler would report an error. At the end of the day our core toolchain works in terms of classpaths, and they are the correct and unique way to identify components in the build.

Reply:
@armanbilge Would you be less opposed to using `project.organization` and `project.name` to align with `sbt` terminology and maybe Mill?

Reply:
sbt and mill are still build tools. I argue that we should treat the scala native compiler toolchain not as a build tool, but as a compiler exactly like the Scala JVM and Scala.js compilers. These compilers operate in terms of classpaths. They do not have any concept of "modules" "projects" "packages" etc.

Reply:
> They do not have any concept of "modules" "projects" "organizations" etc.
This is not really true. A classpath entry is a jar or dir on a file system that represents a project or artifact, that contains packages.

Reply:
> A classpath entry is a jar or dir on a file system that represents a project or artifact

Exactly, and it is identified by its path. It does not have an organization and it does not have a name.